wodenx/Bodiless-JS
# chore(page): Page package

## Overview
This PR contains refactoring for page operations to decouple them from gatsby-theme-bodiless package into a new one. Also, together with the refactoring, we are implementing redirects for both move/delete features.

## Changes
- Created new package named `@bodiless/page`
- Moved jest test case for validatePagePath/validatePageUrl from gatsby theme to bodiless page and updated it with a bit more of coverage cases
- The files PageURLField, PageForm, withClonePageButton, withDeletePageButton, withMovePageButton, withNewPageButton, PageVerification, types, and utils, were moved from gatsby theme to the new page package, reorganized, some renamed, and refactored
- Throughout all files, several duplicates were found and fixed (import the one single function declaration, instead of duplicating it)
- Created constants file to centralize all package constants, instead of having those split across the whole package

## Test Instructions
Because of the refactoring, do regression in the following page features:
- New page
- Clone page
- Move page
- Delete page

Other than that, move/delete should have different fields for redirect option as described in their Github issues.

## Related Issues
- #1099 (redirect on move)
- #1100 (redirect on delete)

## Site Effect
New page depends on Gatsby provider to set the template for the forms. After some discussions, to avoid deps on Gatsby theme bodiless package, we will set temporarily "_default" template and later we will do refactors and provide the template in another way (possibly, we will abstract a PageDataProvider from gatsby). The side effect right is that if a user creates a new page from current page other than using default (like PLP, PDP, etc), it will have "_default" in the template field, instead of having the current template name.
